plugin/Field/SearchcriteriaFields.php

```diff
@@ -102,13 +102,13 @@ public function getFieldLabelsOfInputs(array $inputFormFields): array
 			->addFieldsSearchCriteria($pFieldsCollection)
 			->addFieldsAddressEstate($pFieldsCollection);
 
-		$pGeoFieldsCollection = new FieldModuleCollectionDecoratorGeoPositionFrontend(new FieldsCollection);
-		$pFieldsCollection->merge($pGeoFieldsCollection);
+		$pFieldsCollection->merge(new FieldModuleCollectionDecoratorGeoPositionFrontend(new FieldsCollection));
```

Comment:
This might have a major impact. I hope this works out to be well. The FieldModuleCollectionDecorator* classes were supposed to be decorating the `FieldsCollection` with additional (fictional) fields for a particular purpose, for example the use in a contact form (message is one of those fictional fields). This might unleash fields in areas where they're not supposed to be.

Reply:
This is the reason why `if ($this->_showMessageField)` is needed down below.

Reply:
I think a new `FieldModuleCollectionDecoratorInterestForms` would be the better solution